src/main/scala/org/broadinstitute/hail/driver/Read.scala

```diff
@@ -26,7 +31,42 @@ object Read extends Command {
   def run(state: State, options: Options): State = {
     val input = options.input
 
-    val newVDS = VariantSampleMatrix.read(state.sqlContext, input, options.skipGenotypes)
-    state.copy(vds = newVDS)
+    val inputs = state.hadoopConf.globAll(options.arguments.asScala.toIndexedSeq)
+    if (inputs.isEmpty)
+      fatal("arguments refer to no files")
+
+    val vdses = inputs.map(input => VariantSampleMatrix.read(state.sqlContext, input, options.skipGenotypes))
+
+    val sampleIds = vdses.head.sampleIds
+    val vaSchema = vdses.head.vaSignature
+    val reference = inputs(0)
+
+    vdses.indices.tail.foreach { i =>
+      val ids = vdses(i).sampleIds
+      val vas = vdses(i).vaSignature
+      val path = inputs(i)
+      if (ids != sampleIds) {
+        log.error(s"IDs in reference file $reference: ${ sampleIds.mkString(", ") }")
+        log.error(s"IDs in problem file $path: ${ ids.mkString(", ") }")
+        fatal(
+          s"""cannot read datasets with different sample IDs or sample ordering
+              |  Expected IDs read from $reference
```

Comment:
I'd make the language in the two messages match, e.g., `"Expected IDs read from $reference: ${ sampleIds.mkStr...`.  Also, I'd combine those two `log.error`s into a single one.  We should think of the log as being concurrent and they can get separated.  Same below.